libV2/schemaUtils.js

```diff
@@ -736,6 +736,71 @@ let QUERYPARAM = 'query',
     return schema;
   },
 
+  /** process and resolves types from nested schema structure **/
+
+  processSchema = (resolvedSchema, parentRequired = new Set()) => {
+    if (resolvedSchema.type === 'object' && resolvedSchema.properties) {
+      const schemaDetails = {
+          type: resolvedSchema.type || 'unknown',
+          properties: {},
+          required: []
+        },
+        requiredProperties = new Set(resolvedSchema.required || []);
+
+      for (let [key, prop] of Object.entries(resolvedSchema.properties)) {
+        const propertyDetails = {
+          type: prop.type || 'unknown',
+          deprecated: prop.deprecated || false,
+          enum: prop.enum || undefined,
+          minLength: prop.minLength || undefined,
```

Comment:
P1: Why are we adding these properties always ? Shouldn't we only add these props in case the schema has these attributes defined ?